blocks/video/video.js

```diff
@@ -0,0 +1,374 @@
+
+  return [...scripts].some((script) => script.src === src);
+}
+
+function parseConfig(block) {
+  const isAutoPlay = block.classList.contains('autoplay');
+
+  if (block.classList.contains('hero')) {
+    const videoUrl = block.querySelector(':scope > div > div:first-child').textContent;
+    const title = block.querySelector(':scope > div > div:nth-child(2) > h1')?.textContent;
+    const description = block.querySelector(':scope > div > div:nth-child(2) > p')?.textContent;
+    const button = block.querySelector(':scope > div > div:nth-child(2) > p:last-child')?.textContent;
+
+    return {
+      type: 'hero',
+      videoUrl,
+      isAutoPlay,
+      title,
+      description,
+      button,
+    };
+  }
+
+  if (block.classList.contains('inline')) {
+    const cards = [...block.children].map((child) => {
+      const videoUrl = child.querySelector(':scope > div:first-child').textContent;
+      const title = child.querySelector(':scope > div:nth-child(2) > h1')?.textContent;
+      const description = child.querySelector(':scope > div:nth-child(2) > p')?.textContent;
+
+      return {
+        videoUrl,
+        isAutoPlay,
+        title,
+        description,
+      };
+    });
+
+    return {
+      type: 'cards',
+      cards,
+    };
+  }
+
+  const videoUrl = block.querySelector(':scope div p:first-child').textContent;
+  const posterImage = block.querySelector(':scope div p:nth-child(2)')?.firstElementChild;
+
+  return {
+    type: 'modal',
+    videoUrl,
+    posterImage,
+  };
+}
+
+async function loadVideoJs() {
+  const videoJsScrriptUrl = 'https://vjs.zencdn.net/8.3.0/video.min.js';
+  const videoJsCssUrl = 'https://vjs.zencdn.net/8.3.0/video-js.min.css';
+  if (scriptExists(videoJsScrriptUrl)) {
+    return videoJsScriptPromise;
+  }
+
+  let resolvePromise;
+  videoJsScriptPromise = new Promise((resolve) => {
+    resolvePromise = resolve;
+  });
+
+  const css = document.createElement('link');
+  css.setAttribute('href', videoJsCssUrl);
+  css.setAttribute('rel', 'stylesheet');
+
+  const mainScript = document.createElement('script');
+  mainScript.setAttribute('src', videoJsScrriptUrl);
+  mainScript.setAttribute('async', 'true');
+  mainScript.onload = () => resolvePromise();
+
+  const header = document.querySelector('head');
+  header.append(css);
+  header.append(mainScript);
+
+  return videoJsScriptPromise;
+}
+
+function createPlayButton(container, player) {
+  const pauseIcon = document.createElement('span');
+  pauseIcon.classList.add('icon');
+  pauseIcon.classList.add('icon-pause');
+
+  const playIcon = document.createElement('span');
+  playIcon.classList.add('icon');
+  playIcon.classList.add('icon-play');
+
+  function updateIcons(isPaused) {
+    if (isPaused) {
+      playIcon.style.display = '';
+      pauseIcon.style.display = 'none';
+    } else {
+      playIcon.style.display = 'none';
+      pauseIcon.style.display = '';
+    }
+  }
+
+  const button = document.createElement('button');
```

Comment:
please ensure all interactive elements (e.g., buttons) are accessible, with appropriate ARIA attributes where necessary.

Reply:
Yes. I will be working on accessibility next